responsibleai_vision/responsibleai_vision/managers/explainer_manager.py

```diff
@@ -249,16 +257,22 @@ def compute_single_explanation(self,
                                                          self._classes)
                     else:
                         self._model = PytorchDRiseWrapper(self._model._model,
-                                                          len(self._classes))
+                                                          len(self._classes),
+                                                          self._device)
 
                 # calling DRISE to generate saliency maps for all objects
                 mask_res_tuple = (self._mask_res, self._mask_res)
+                device = self._device
+                # get_drise_saliency_map only recognizes GPU and CPU
+                if device == Device.AUTO.value:
+                    device = None
                 fl, _, _, = get_drise_saliency_map(img,
                                                    self._model,
                                                    len(self._classes),
                                                    savename=str(index),
                                                    nummasks=self._num_masks,
                                                    maskres=mask_res_tuple,
+                                                   devicechoice=device,
```

Comment:
What if `None` is passed here for device after line 268?

Reply:
see https://github.com/microsoft/vision-explanation-methods/blob/657fe461264a596c94378a6a1174a3c21d0757fe/python/vision_explanation_methods/DRISE_runner.py#L117

it just uses same logic as auto:

torch.device('cuda' if torch.cuda.is_available() else 'cpu')